### PR TITLE
Add timeout to groupby and sorter result processors

### PR DESCRIPTION
**Pull request by ashtul**
Before this PR, Timeout was set only on the index result processor (`indexer`) which holds all inverted index iterators.
Two result processors, `sorter` and `grouper` have internal storage where they collect all results from the `indexer`.  Before this fix, if a timeout is reached while results are being read from a sorter or grouper, the query would not stop.

### CODE REVIEW

File: src/result_processor.c [hunk 1]
```diff
@@ -68,7 +68,7 @@ static int rpidxNext(ResultProcessor *base, SearchResult *res) {
   RPIndexIterator *self = (RPIndexIterator *)base;
   IndexIterator *it = self->iiter;
 
-  if (++self->timeoutLimiter == 100) {
+  if (++self->timeoutLimiter == DEFAULT_TIMEOUT_LIMIT) {
```
Comment (oshadmi) [inline src/result_processor.c:71]: Why change the count from 100 to 1000? (less accurate timeout, but more efficient?)

**Review by oshadmi — APPROVED** [review]
LGTM 👍🏼 
(Just one question)

File: src/result_processor.c (hunk 1 above)
Comment (ashtul) [reply src/result_processor.c:71]: diff in execution time is minimal to it won't lengthen the timeout by anything significant.
@MeirShpilraien WDYT?

**Review by oshadmi — APPROVED** [review]
LGTM👍🏼

**Pull request merged**